build_tools/kokoro/gcp_ubuntu/cmake/linux/x86-swiftshader-asan/build.sh

```diff
@@ -126,7 +133,10 @@ excluded_tests_regex="($(IFS="|" ; echo "${excluded_tests[*]?}"))"
 
 cd ${CMAKE_BUILD_DIR?}
 
-echo "Testing with ctest"
+echo "******************** Running main project ctests ************************"
 ctest --timeout 900 --output-on-failure \
   --label-exclude "${label_exclude_regex}" \
   --exclude-regex "${excluded_tests_regex?}"
+
+echo "******************** llvm-external-projects tests ***********************"
+cmake --build . --target check-iree-dialects -- -k 0
```

Comment:
https://source.cloud.google.com/results/invocations/1a416c88-92ac-447f-8010-e5444a4d82dd/targets/iree%2Fgcp_ubuntu%2Fcmake%2Flinux%2Fx86-swiftshader-asan%2Fpresubmit/log

```
Failed Tests (4):
  IREE_DIALECTS :: Dialect/iree_linalg_ext/fuse-operands.mlir
  IREE_DIALECTS :: Dialect/linalg_transform/tile-and-peel.mlir
  IREE_DIALECTS :: Dialect/linalg_transform/tile.mlir
  IREE_DIALECTS :: iree_pydm/rtl/link.mlir


Testing Time: 3.53s
  Unsupported      : 13
  Passed           : 51
  Expectedly Failed:  1
  Failed           :  4
```

None of those failures are immediately obvious to me from the logs. All look like failures in `FileCheck` matching, not issues reported by ASan. I'd also like to test that the ASan reporting does something, probably by deliberately introducing a bug and checking the logs.

I'm looking into how to exclude those 4 tests for now. They are following LLVM style and using `cmake --build` instead of `ctest`, so maybe exclude via the lit runner in some way?

Reply:
Okay, testing locally and I _think_ those four tests actually fail when asserts are not enabled.
This attribute gets added only in debug builds, and the lit tests are checking for it: https://github.com/google/iree/blob/b1e8b64b07fe60bc83bdc2fa59c00aed44ad5978/llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/PDL.cpp#L52-L57

Reply:
(At least for `Dialect/linalg_transform/tile.mlir` - still need to triage the other tests, `iree_pydm/rtl/link.mlir` may be different)

Reply:
Fixed the linalg tests to pass with and without asserts / debug mode enabled.

All that's left for enabling ASan coverage of `check-iree-dialects` is figuring out the `iree_pydm/rtl/link.mlir` failure. That test depends on the Python bindings being enabled, which is not compatible with ASan.

error: `<unknown>:0: error: unexpected error: could not open input file /home/kbuilder/build/third_party/llvm-project/llvm/tools/mlir-iree-dialects/resources/PyDMRTL/PyDMRTLBase.mlir`
https://github.com/google/iree/blob/aae314e8bb72115aa00ce2556fbd8ed1559d459d/llvm-external-projects/iree-dialects/resources/CMakeLists.txt#L1-L3 https://github.com/google/iree/blob/aae314e8bb72115aa00ce2556fbd8ed1559d459d/llvm-external-projects/iree-dialects/resources/PyDMRTL/CMakeLists.txt#L34-L38

can we conditionally exclude https://github.com/google/iree/tree/main/llvm-external-projects/iree-dialects/test/iree_pydm from https://github.com/google/iree/blob/main/llvm-external-projects/iree-dialects/test/CMakeLists.txt, @stellaraccident ?

Reply:
Ah, found https://github.com/google/iree/blob/main/llvm-external-projects/iree-dialects/test/python/lit.local.cfg. Giving that a try.